ntoskrnl/include/internal/ps_i.h

```diff
@@ -5,4 +5,585 @@
+    (
+        ULONG,
+        ULONG,
+        ICIF_QUERY | ICIF_SET | ICIF_SIZE_VARIABLE
+    ),
+
+    /* ProcessDebugObjectHandle */
+    IQS_SAME
+    (
+        HANDLE,
+        ULONG,
+        ICIF_QUERY | ICIF_QUERY_SIZE_VARIABLE
+    ),
+
+    /* ProcessDebugFlags */
+    IQS_SAME
+    (
+        ULONG,
+        ULONG,
+        ICIF_QUERY | ICIF_SET | ICIF_SIZE_VARIABLE
+    ),
+
+    /* ProcessHandleTracing */
+    IQS_SAME
+    (
+        ULONG,
+        ULONG,
+        ICIF_QUERY | ICIF_SET | ICIF_SIZE_VARIABLE
+    ),
+
+    /* ProcessIoPriority */
+    IQS_SAME
+    (
+        ULONG,
+        ULONG,
+        ICIF_QUERY | ICIF_SET | ICIF_SIZE_VARIABLE
+    ),
+
+    /* ProcessExecuteFlags */
+    IQS_SAME
+    (
+        ULONG,
+        ULONG,
+        ICIF_QUERY | ICIF_SET | ICIF_SIZE_VARIABLE
+    ),
+
+    /* ProcessTlsInformation */
+    IQS_SAME
+    (
+        ULONG,
+        ULONG,
+        ICIF_QUERY | ICIF_SET | ICIF_SIZE_VARIABLE
+    ),
+
+    /* ProcessCookie */
+    IQS_SAME
+    (
+        ULONG,
+        ULONG,
+        ICIF_QUERY | ICIF_QUERY_SIZE_VARIABLE
+    ),
+
+    /* ProcessImageInformation */
+    IQS_SAME
+    (
+        SECTION_IMAGE_INFORMATION,
+        ULONG,
+        ICIF_QUERY | ICIF_QUERY_SIZE_VARIABLE
+    ),
+
+    /* ProcessCycleTime */
+    IQS_SAME
+    (
+        /*
+         * FIXME: For the time being, use ULONG as
+         * PROCESS_CYCLE_TIME_INFORMATION structure is not
+         * declared in the headers yet.
+         */
+        ULONG,
+        ULONG,
+        ICIF_QUERY | ICIF_SET | ICIF_SIZE_VARIABLE
+    ),
+
+    /* ProcessPagePriority */
+    IQS_SAME
+    (
+        ULONG,
+        ULONG,
+        ICIF_QUERY | ICIF_SET | ICIF_QUERY_SIZE_VARIABLE
+    ),
+
+    /* ProcessInstrumentationCallback */
+    IQS_SAME
+    (
+        ULONG,
+        ULONG,
+        ICIF_QUERY | ICIF_SET | ICIF_SIZE_VARIABLE
+    ),
+};
+
+//
+// Thread Information Classes
+//
+static const INFORMATION_CLASS_INFO PsThreadInfoClass[] =
+{
+    /* ThreadBasicInformation */
+    IQS_SAME
+    (
+        THREAD_BASIC_INFORMATION,
+        ULONG,
+        ICIF_QUERY | ICIF_QUERY_SIZE_VARIABLE
+    ),
+
+    /* ThreadTimes */
+    IQS_SAME
+    (
+        KERNEL_USER_TIMES,
+        ULONG,
+        ICIF_QUERY | ICIF_QUERY_SIZE_VARIABLE
+    ),
+
+    /* ThreadPriority */
+    IQS_SAME
+    (
+        KPRIORITY,
+        ULONG,
+        ICIF_SET | ICIF_SET_SIZE_VARIABLE
+    ),
+
+    /* ThreadBasePriority */
+    IQS_SAME
+    (
+        LONG,
+        ULONG,
+        ICIF_SET | ICIF_SET_SIZE_VARIABLE
+    ),
+
+    /* ThreadAffinityMask */
+    IQS_SAME
+    (
+        KAFFINITY,
+        ULONG,
+        ICIF_SET | ICIF_SET_SIZE_VARIABLE
+    ),
+
+    /* ThreadImpersonationToken */
+    IQS_SAME
+    (
+        HANDLE,
+        ULONG,
+        ICIF_SET | ICIF_SET_SIZE_VARIABLE
+    ),
+
+    /* ThreadDescriptorTableEntry is only implemented in x86 as well as the descriptor entry */
+    #if defined(_X86_)
```

Comment:
This entry can't just be if'd out, that would move all the information classes behind it to the front. So for other arches there needs to be an IQS_NONE entry